src/main/java/de/tum/in/www1/artemis/web/rest/FileUploadSubmissionResource.java

```diff
@@ -251,9 +250,10 @@ else if (!authCheckService.isAtLeastInstructorForExercise(exercise)) {
             return badRequest();
         }
 
-        // Tutors cannot start assessing submissions if the exercise due date hasn't been reached yet
-        if (fileUploadExercise.getDueDate() != null && fileUploadExercise.getDueDate().isAfter(ZonedDateTime.now())) {
-            return notFound();
+        // Check if tutors can start assessing the students submission
+        boolean startAssessingSubmissions = this.examSubmissionService.checkIfExerciseDueDateIsReached(fileUploadExercise);
```

Comment:
While this seems to be a correct implementation, it is hard to follow that this method does a useful check for non-exam exercises. See further comment in `ExamSubmissionService`.